src/plugins/embeddable/public/lib/state_transfer/embeddable_state_transfer.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { cloneDeep } from 'lodash';
+import { CoreStart, ScopedHistory } from '../../../../../core/public';
+import {
+  EmbeddableOriginatingAppState,
+  isEmbeddableOriginatingAppState,
+  EmbeddablePackageState,
+  isEmbeddablePackageState,
+} from './types';
+
+export class EmbeddableStateTransfer {
+  constructor(private coreStart: CoreStart) {}
```

Comment:
To eliminate tight coupling to core, I would only pass in here what is necessary - `this.coreStart.application.navigateToApp`

Recommendation not merged yet but can see a draft here:
https://github.com/elastic/kibana/pull/66266/files#diff-c317fe5764b7c4dd7eb51373b9d7c45eR1

Reply:
Changed! Nice doc